backend/modules/pdf_processing/repository.py

```
@@ -0,0 +1,352 @@
+            
+            all_personas_final = self.db.query(ScenarioPersona).filter(
+                ScenarioPersona.scenario_id == scenario.id,
+                ScenarioPersona.deleted_at.is_(None)
+            ).all()
+            has_personas_with_images = any(persona.image_url for persona in all_personas_final)
+            
+            scenario.images_completed = has_scenes_with_images or has_personas_with_images
+            
+            self.db.commit()
+            debug_log(f"[REPOSITORY] Successfully saved full data for scenario {scenario_id}")
+            return True
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)